google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java

```diff
@@ -517,7 +524,12 @@ private AppendRequestAndResponse pollInflightRequestQueue() {
     return new StreamWriter.Builder(streamName);
   }
 
-  /** A builder of {@link StreamWriterV2}s. */
+  /** Thread-safe getter of updated TableSchema */
+  public synchronized TableSchema getUpdatedSchema() {
```

Comment:
Since it is actually guarded by lock, so just use this.lock.lock() in the method body.

Reply:
Done.